### PR TITLE
Support `$rename` field update operator

### LINKED ISSUE #626: Support `$rename` field update operator
https://www.mongodb.com/docs/manual/reference/operator/update/rename/

### DoD

Dot notation should be supported and tested. We should also check if we need to add test cases for dot notation to other compat update tests.

Update documentation about supported operators.

### PR DESCRIPTION
**Pull request by fenogentov**
Closes #626.

### CODE REVIEW

**Comment by AlekSi** [comment]
@fenogentov I will do a real review later, but please golangci-lint errors in the meantime. We enabled wsl linter for the new code.

File: integration/helpers.go [hunk 1]
```diff
@@ -168,9 +168,9 @@ func AssertEqualError(t testing.TB, expected mongo.CommandError, actual error) b
 //
 // In general, error messages should be the same. Exceptions include:
 //
-//  * MongoDB typos (e.g. "sortto" instead of "sort to");
-//  * MongoDB values formatting (e.g. we don't want to write additional code to format
-//    `{ $slice: { a: { b: 3 }, b: "string" } }` exactly the same way).
+//   * MongoDB typos (e.g. "sortto" instead of "sort to");
```
Comment (AlekSi) [inline integration/helpers.go:171]: Moved that to #1003 to make this PR smaller

**Review by AlekSi — COMMENTED** [review]
There are conflicts now

**Comment by AlekSi** [comment]
@fenogentov Hi! Was it intentional, or you force-pushed by mistake?

**Comment by fenogentov** [comment]
> @fenogentov Hi! Was it intentional, or you force-pushed by mistake?

I do not understand what happened
  I didn't press anything

**Pull request closed without merging**